src/System.Windows.Forms/src/System/Windows/Forms/ListView.ListViewItemAccessibleObject.cs

```diff
@@ -0,0 +1,372 @@
+                    UiaCore.UIA.IsOffscreenPropertyId => (State & AccessibleStates.Offscreen) == AccessibleStates.Offscreen,
+                    UiaCore.UIA.NativeWindowHandlePropertyId => _owningListView.Handle,
+                    UiaCore.UIA.IsSelectionItemPatternAvailablePropertyId => IsPatternSupported(UiaCore.UIA.SelectionItemPatternId),
+                    UiaCore.UIA.IsScrollItemPatternAvailablePropertyId => IsPatternSupported(UiaCore.UIA.ScrollItemPatternId),
+                    UiaCore.UIA.IsInvokePatternAvailablePropertyId => IsPatternSupported(UiaCore.UIA.InvokePatternId),
+                    _ => base.GetPropertyValue(propertyID)
+                };
+
+            /// <summary>
+            ///  Indicates whether specified pattern is supported.
+            /// </summary>
+            /// <param name="patternId">The pattern ID.</param>
+            /// <returns>True if specified </returns>
+            internal override bool IsPatternSupported(UiaCore.UIA patternId)
+            {
+                if (patternId == UiaCore.UIA.ScrollItemPatternId ||
+                    patternId == UiaCore.UIA.LegacyIAccessiblePatternId ||
+                    patternId == UiaCore.UIA.SelectionItemPatternId ||
+                    patternId == UiaCore.UIA.InvokePatternId)
+                {
+                    return true;
+                }
+
+                return base.IsPatternSupported(patternId);
+            }
+
+            internal override void RemoveFromSelection()
+            {
+                // Do nothing, C++ implementation returns UIA_E_INVALIDOPERATION 0x80131509
+            }
+
+            internal override UiaCore.IRawElementProviderSimple ItemSelectionContainer => _owningListView.AccessibilityObject;
+
+            internal override void ScrollIntoView()
+            {
+                int currentIndex = CurrentIndex;
+
+                if (_owningListView.SelectedItems == null) //no items selected
+                {
+                    User32.SendMessageW(_owningListView, (User32.WM)User32.LB.SETCARETINDEX, (IntPtr)currentIndex);
+                    return;
+                }
+
+                int firstVisibleIndex = User32.SendMessageW(_owningListView, (User32.WM)User32.LB.GETTOPINDEX).ToInt32();
+                if (currentIndex < firstVisibleIndex)
+                {
+                    User32.SendMessageW(_owningListView, (User32.WM)User32.LB.SETTOPINDEX, (IntPtr)currentIndex);
+                    return;
+                }
+
+                int itemsHeightSum = 0;
+                int visibleItemsCount = 0;
+                int listBoxHeight = _owningListView.ClientRectangle.Height;
+                int itemsCount = _owningListView.Items.Count;
+
+                for (int i = firstVisibleIndex; i < itemsCount; i++)
+                {
+                    int itemHeight = User32.SendMessageW(_owningListView, (User32.WM)User32.LB.GETITEMHEIGHT, (IntPtr)i).ToInt32();
```

Comment:
```suggestion
                    int itemHeight = PARAM.ToInt(User32.SendMessageW(_owningListView, (User32.WM)User32.LB.GETITEMHEIGHT, (IntPtr)i));
```

Applies to other instances as well

Reply:
resolved